builder/builder.go

```diff
@@ -62,91 +87,60 @@ func FromBuild(build *api.Build) Builder {
 	}
 }
 
-// NewBuild is the API entry for the BuilderScheduler. Create a new Build instance based on PlatformBuild.
-func NewBuild(platformBuild api.PlatformBuild, publishImage string, buildName string) Scheduler {
-	// TODO: Figure if we need a PlatformBuild builder fluent api.
-	// TODO: Verify structure integrity
+// NewBuild is the API entry for the Builder. Create a new Build instance based on PlatformBuild.
+func NewBuild(info BuilderInfo) Scheduler {
 	ctx := buildContext{
-		BaseImage: platformBuild.Spec.BaseImage,
+		BaseImage: info.Platform.Spec.BaseImage,
 		C:         context.TODO(),
 	}
 
-	if platformBuild.Spec.BuildStrategy == api.BuildStrategyPod && platformBuild.Spec.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
-		ctx.Build = api.NewKanikoBuild(platformBuild, publishImage, buildName)
-	} else {
-		panic(fmt.Errorf("BuildStrategy %s with PublishStrategy %s is not supported", platformBuild.Spec.BuildStrategy, platformBuild.Spec.PublishStrategy))
+	for _, v := range schedulers {
+		if v.CanHandle(info) {
+			return v.CreateScheduler(info, ctx)
+		}
 	}
+	panic(fmt.Errorf("BuildStrategy %s with PublishStrategy %s is not supported", info.Platform.Spec.BuildStrategy, info.Platform.Spec.PublishStrategy))
```

Comment:
what happens if the issue is not related with the Build and Publish strategy?
maybe handle the error in the CabHandle method.

Reply:
It's. If there's no suitable builder available, we panic. That's why we have this here. Any other issue will panic before this call.